org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/DeviceRequestHandler.java

```diff
@@ -1,249 +1,249 @@
-			}
-			addDeviceInformationAndAttributes(device, request);
-
-		} else {  // Device list needed.
-
-			Collection<DeviceInformation<?>> info;
-			if (request.isIncludeNonAlive()) {
-				info = dservice.getDeviceInformationIncludingNonAlive();
-			} else {
-				info = dservice.getDeviceInformation();
-			}
-			request.setDevices(info);
-		}
-	}
-
-	private static void addDeviceInformationAndAttributes(
-			IRunnableDevice<Object> device, DeviceRequest request) throws ScanningException {
-		// get the device information and set it in the request
-		DeviceInformation<?> info = ((AbstractRunnableDevice<?>)device).getDeviceInformation();
-		request.addDeviceInformation(info);
-
-		// check this device can have attribute, if so cast it
-		if (!(device instanceof IAttributableDevice)) return;
-		IAttributableDevice attrDevice = (IAttributableDevice) device;
-
-		if (request.isGetAllAttributes()) {
-			// convert the list of attributes to a map and set in the request
-			request.setAttributes(attrDevice.getAllAttributes().stream()
-					.collect(toMap(IDeviceAttribute::getName, identity())));
-		} else if (request.getAttributeName() != null) {
-			// add the single attribute with the given name - note: exception thrown if no such attribute
- 			request.addAttribute(attrDevice.getAttribute(request.getAttributeName()));
-		}
-	}
-
-}
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.api.device;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Collection;
+
+import org.eclipse.scanning.api.INameable;
+import org.eclipse.scanning.api.IScannable;
+import org.eclipse.scanning.api.ITerminatable;
+import org.eclipse.scanning.api.ModelValidationException;
+import org.eclipse.scanning.api.MonitorRole;
+import org.eclipse.scanning.api.annotation.ui.DeviceType;
+import org.eclipse.scanning.api.event.EventException;
+import org.eclipse.scanning.api.event.core.IPublisher;
+import org.eclipse.scanning.api.event.core.IRequestHandler;
+import org.eclipse.scanning.api.event.scan.DeviceAction;
+import org.eclipse.scanning.api.event.scan.DeviceInformation;
+import org.eclipse.scanning.api.event.scan.DeviceRequest;
+import org.eclipse.scanning.api.malcolm.attributes.IDeviceAttribute;
+import org.eclipse.scanning.api.scan.ScanningException;
+
+/**
+ * TODO FIXME Is the idea of having request/response calls correct for exposing
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)